code/modules/food/food/condiment.dm

```diff
@@ -441,30 +453,87 @@
 
 //End of MRE stuff.
 
-/obj/item/weapon/reagent_containers/food/condiment/flour
-	name = "flour sack"
-	desc = "A big bag of flour. Good for baking!"
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour
+	name = "flour carton"
+	desc = "A big carton of flour. Good for baking!"
 	icon = 'icons/obj/food.dmi'
 	icon_state = "flour"
 	volume = 220
 	center_of_mass = list("x"=16, "y"=8)
 
-/obj/item/weapon/reagent_containers/food/condiment/flour/on_reagent_change()
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/on_reagent_change()
+	update_icon()
 	return
 
-/obj/item/weapon/reagent_containers/food/condiment/flour/Initialize()
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/Initialize()
 	. = ..()
 	reagents.add_reagent("flour", 200)
 	randpixel_xy()
 
+/obj/item/weapon/reagent_containers/food/condiment/advanced/update_icon()
+	overlays.Cut()
+
+	if(reagents.total_volume)
+		var/image/filling = image('icons/obj/food.dmi', src, "[icon_state]10")
+
+		var/percent = round((reagents.total_volume / volume) * 100)
+		switch(percent)
+			if(-1 to 9)		filling.icon_state = "[icon_state]-10"
+			if(10 to 24) 	filling.icon_state = "[icon_state]-25"
+			if(25 to 49)	filling.icon_state = "[icon_state]-50"
+			if(50 to 85)	filling.icon_state = "[icon_state]-75"
+			if(86 to INFINITY)	filling.icon_state = "[icon_state]-100"
+
+		overlays += filling
+
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/rustic
+	name = "flour sack"
+	desc = "An artisanal sack of flour. Classy!"
+	icon_state = "flour_bag"
+
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/rustic/empty
+
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/rustic/empty/Initialize()
+	. = ..()
+	reagents.remove_reagent("flour", 200)
+	randpixel_xy()
+
+/obj/item/weapon/reagent_containers/food/condiment/advanced/sugar
+	name = "sugar carton"
+	desc = "A big carton of sugar. Sweet!"
+	icon = 'icons/obj/food.dmi'
+	icon_state = "sugar"
+	volume = 110
+	center_of_mass = list("x"=16, "y"=8)
+
+/obj/item/weapon/reagent_containers/food/condiment/advanced/sugar/on_reagent_change()
+	update_icon()
+	return
+
+/obj/item/weapon/reagent_containers/food/condiment/advanced/sugar/Initialize()
+	. = ..()
+	reagents.add_reagent("sugar", 100)
+	randpixel_xy()
+
+/obj/item/weapon/reagent_containers/food/condiment/advanced/sugar/rustic
+	name = "sugar sack"
+	desc = "An artisanal sack of sugar. Classy!"
+	icon_state = "sugar_bag"
+
+/obj/item/weapon/reagent_containers/food/condiment/advanced/sugar/rustic/empty
+
```

Comment:
```suggestion
```
You don't have to declare the subtype like this for it to exist.

Reply:
Don't I need the path to exist to allow empty bags to spawn in the botanist lockers?

Reply:
By making a proc on the subtype (Which you do immediately below), the path exists in the class tree. If you hadn't declared any logic for the `advanced` type, but declared the flour and sugar cartons the same, they'd still exist and be things that you can spawn in, and there'd also be an `advanced` type (minus the /flour or /sugar subtyping).